feat: match `pause_menu.c`

moisesPC opened the pull request.

moisesPC committed:
Add addresses

moisesPC committed:
First changes to C file

moisesPC committed:
Add headers

moisesPC committed:
Match bekkan_1f.c

moisesPC committed:
Docs

moisesPC committed:
Add obj 17c addresses

moisesPC committed:
Add headers obj17C

moisesPC committed:
Match obj 17c code

moisesPC committed:
Match obj 17C data

moisesPC committed:
Match tou_turo.c .data

moisesPC committed:
format

moisesPC committed:
Refactor mfds with new discoveries

moisesPC committed:
comment

moisesPC committed:
Refactor unrenamed actor settings -> actor config for consistency

moisesPC committed:
More deeply document actor config

moisesPC committed:
format

moisesPC committed:
clarification

moisesPC committed:
Add gameplayMenuMgr function prototypes and misc refactor

moisesPC committed:
typo

moisesPC committed:
Misc changes

moisesPC committed:
Prepare and start gameplay_menu_mgr.c. Match data and bss

moisesPC committed:
Add most of the code

moisesPC committed:
Fixed matching

moisesPC committed:
temporary clang format off

moisesPC committed:
Match rodata

moisesPC committed:
misc

moisesPC committed:
gameplayMenuMgr_insideMenuLoop

moisesPC committed:
temporary clang

moisesPC committed:
Docs, addresses and temporarily remove rodata

moisesPC committed:
Remove data string guesser and add rodata shift-jis encoding for analyzer

moisesPC committed:
Link makerom

moisesPC committed:
Revert to previous state due to building problems

moisesPC committed:
Initial gameplay_menu_mgr_textbox.c setup

moisesPC committed:
Match gameplay_menu_mgr_textbox.c

moisesPC committed:
Refactor proposal

moisesPC committed:
Match mini_scroll_setup.c

moisesPC committed:
Fix name

moisesPC committed:
Document gameplay_common_textbox.c

moisesPC committed:
Document mini scroll

moisesPC committed:
Add driver code function prototypes for mini scroll

moisesPC committed:
Create entrance map name display C

moisesPC committed:
Add code and rodata

moisesPC committed:
Add addresses and match code and rodata

moisesPC committed:
Match data

moisesPC committed:
Docs

moisesPC committed:
Small clarifications

moisesPC committed:
More notes

moisesPC committed:
Corrected buffer size

moisesPC committed:
Start enemy_list.c

moisesPC committed:
enemy list progress

moisesPC committed:
Match .bss

moisesPC committed:
Docs

moisesPC committed:
Doc EnemyList_getNumberOfActiveEnemies

moisesPC committed:
Remove null bytes from EnemyList_enemyStates and added them into a separate array

moisesPC committed:
Add a couple of mfds flags

moisesPC committed:
Finish enemy_list.c

moisesPC committed:
Link debug.c bss

moisesPC committed:
Readded comment that was accidentally deleted

moisesPC committed:
Refactor effect.h

moisesPC committed:
Remove extern from debug.h

moisesPC committed:
More effect work

moisesPC committed:
Introduce fire

moisesPC committed:
Small fixes to effect.h

moisesPC committed:
Start fire.c

moisesPC committed:
Add fire sparkes header + corrected return types

moisesPC committed:
Complete fire sparkles header

moisesPC committed:
Small fixes

moisesPC committed:
Some fire.c improvements

moisesPC committed:
Add fire sparkles

moisesPC committed:
Add non matching code

moisesPC committed:
Small corrections

moisesPC committed:
Match FireSparkles_Init

moisesPC committed:
Addresses for pause menu

moisesPC committed:
Create headers and C file

moisesPC committed:
Match first few functions

moisesPC committed:
Match pauseMenu_updateClock

moisesPC committed:
Correct return types

moisesPC committed:
Correct data types

moisesPC committed:
Match pauseMenu_createItemDescription

moisesPC committed:
Temporarily add ifdef until we start linking rodata from the C file

moisesPC committed:
Match pauseMenu_checkScrollObjExists

moisesPC committed:
Fill out some sound menu work variables

moisesPC committed:
Match (but not link yet) pauseMenu_createSoundMenuWork

moisesPC committed:
Match pauseMenu_createMainMenu

moisesPC committed:
Add pause menu func id enums

moisesPC committed:
Replace literals with enum values

moisesPC committed:
pauseMenu_calcItemList

moisesPC committed:
Introduce union in Model struct

moisesPC committed:
Reorganize a few includes

moisesPC committed:
Reorganize other includes

moisesPC committed:
Try to fix circular dependency

moisesPC committed:
Match pauseMenu_calcQuitMenu

moisesPC committed:
Current progress

moisesPC committed:
Small fixes

moisesPC committed:
Match .data

moisesPC committed:
Small doc

moisesPC committed:
Refactor sound_menu_work -> pause_item_menu_work

moisesPC committed:
Start necro

moisesPC committed:
Match first few necro functions

moisesPC committed:
Match necro_finishedClosing

moisesPC committed:
Remove comma

moisesPC committed:
Match necro_init

moisesPC committed:
Add funcs Id

moisesPC committed:
Corrected data type

moisesPC committed:
necro_close

moisesPC committed:
necro_loop

moisesPC committed:
necro docs

moisesPC committed:
Fix flag assignment

moisesPC committed:
Page refactor

moisesPC committed:
Match and link pause_menu rodata + Match pauseMenu_init

moisesPC committed:
Fix necro page array size

moisesPC committed:
Small fix

moisesPC committed:
Match pauseMenu_checkIfItemCanBeUsed and pauseMenu_calcMainMenu

moisesPC committed:
Match pauseMenu_updateDigitalClockDisplay

moisesPC committed:
pauseMenu_calcItemSelectedMenu

moisesPC committed:
pause menu docs part 1-?

moisesPC committed:
Remove unused variable

moisesPC committed:
pause menu docs part 2-? Macros to help with readability

moisesPC committed:
pause menu docs 3-? pauseMenu_calcQuitMenu and MiniScrollParameters docs

moisesPC committed:
pause menu docs 4-? First comments pass done

moisesPC committed:
Fix mistake

moisesPC committed:
pause menu docs 5-?

moisesPC committed:
Fix mistake in comment

blazkowolf committed:
fix: upgrade mise tasks to use new features

blazkowolf commented on src/overlay/entrance_map_name_display.c:
```diff
@@ -0,0 +1,311 @@
+ * @note The X position for each vertex is determined
+ *       in function `entranceMapNameDisplay_init`
+ */
+Vtx entranceMapNameDisplay_bgModelVertices[] = {
```
Ideally, these vertices should not be checked into the repo and be included from the assets directory instead. Is there a technical reason this was inlined in the code?

blazkowolf replied:
@moisesPC

blazkowolf commented on src/overlay/entrance_map_name_display.c:
```diff
@@ -0,0 +1,311 @@
+/**
+ * First material assigned to the background model
+ */
+Gfx entranceMapNameDisplay_bgModelMaterial1[] = {
```
Same question here for this display list.

blazkowolf commented on src/overlay/entrance_map_name_display.c:
```diff
@@ -0,0 +1,311 @@
+/**
+ * Second material assigned to the background model
+ */
+Gfx entranceMapNameDisplay_bgModelMaterial2[] = {
```
Same question here for this display list.

blazkowolf committed:
refactor(entrance_map_name_display): fixup some control flow

blazkowolf committed:
docs: update LZKN64 link

blazkowolf committed:
fix(entrance_map_name_display): introduce enum

Exchange `ENTRANCE_MAP_NAME_` defines for an enum
and move them from `cv64.h` into their appropriate
object header.

blazkowolf committed:
refactor(entrance_map_name_display): rename `{e,E}ntranceMapNameDisplay`

blazkowolf committed:
style(actor_config): abide 100 column limit

blazkowolf committed:
refactor(debug): use existing enum

Use `DebugProcessMeterState` enum in `processMeter_update`.

blazkowolf committed:
fix(figure): introduce enum

Introduce `FigureFlag` enum in place of defines.

blazkowolf committed:
fix(model): resolve `miniScroll.h` recursive dep

blazkowolf committed:
fix(pause_item_menu_work): add include

Add `#include` for `mfds.h` since `MfdsState` is used.

blazkowolf committed:
refactor(object): rename `cv64_object_exec_flag_t->ObjectExecFlag`

blazkowolf committed:
refactor(object): rename `cv64_object_func_inf_t->ObjectFuncInfo`

blazkowolf committed:
refactor(object): rename `cv64_object_file_inf_t->ObjectFileInfo`

blazkowolf committed:
refactor(object): rename `cv64_object_func_t->ObjectFunc`

blazkowolf committed:
refactor: rename a lot of stuff

1. `masterLightMgr.h->master_light_manager.h`
2. `modelLighting.h->model_lighting.h`
3. `pointLight.h->point_light.h`

blazkowolf committed:
build(tools): use Torch CMake config for building

Invoke CMake to build Torch instead of invoking Make
inside our `CMakeLists.txt` file to build Torch. This
is more flexible because it allows the user to pick
whichever build system they prefer.

blazkowolf committed:
refactor(effect): rename fire func types

blazkowolf committed:
refactor(effect): rename `pickableItemFlash->PickableItemFlash`

blazkowolf committed:
refactor(overlay): rename `openingBat->OpeningBat`

blazkowolf committed:
refactor(map): rename `DecorativeChandelierFunc{s,}`

blazkowolf committed:
amend! refactor(effect): rename `{p,P}ickableItemFlash`

refactor(effect): rename `pickableItemFlash->PickableItemFlash`

blazkowolf committed:
amend! refactor(overlay): rename `{o,O}peningBat`

refactor(overlay): rename `openingBat->OpeningBat`

blazkowolf committed:
refactor(overlay): rename `Bekkan1FSquareFunc{s,}`

blazkowolf committed:
refactor(object): rename `{o,O}bj0172Func{s,}`

blazkowolf committed:
refactor(object): rename `{o,O}bj017CFunc{s,}`

blazkowolf committed:
refactor(object): rename `character{S,_s}elect.h`

blazkowolf committed:
refactor(overlay): rename `{e,E}ntranceMapNameDisplay`

blazkowolf committed:
refactor(object): rename various `gameplayMenuMgr` things

blazkowolf commented on include/game/objects/menu/mfds.h:
```diff
@@ -149,21 +171,30 @@ typedef struct MfdsColorAnimationState {
+     * For the current printing character, the first byte is the Y pos offset,
+     * and the second byte is the X pos offset
+     */
+    s16 Y_and_X_pos_offsets;
```
@moisesPC I wonder if fields like this could be a struct of two `s8`s?

blazkowolf commented on include/game/objects/menu/mfds.h:
```diff
@@ -314,7 +361,7 @@ text_convertSignedIntegerToText(u32 number, u16* dest, u8 number_of_chars, u32 n
 extern u16* convertUTF16ToCustomTextFormat(u16* text_buffer);
 extern void textbox_setHeightAndWidth(MfdsState* self, u32 index, u8 text_height, u8 text_width);
-extern MfdsState* map_getMessageFromPool(u16 text_ID, u8 textbox_display_time);
+extern MfdsState* gameplayCommonTextbox_getMapMessage(u16 text_ID, u8 textbox_display_time);
```
@moisesPC should this function be defined with the other `gameplayCommonTextbox` functions in `gameplay_menu_manager.h`?

blazkowolf committed:
refactor: use `for` loops in `enemy_list.c`

blazkowolf committed:
refactor(object): rename `mini{S,_s}croll.h`

blazkowolf committed:
refactor(object): rename `{m,M}iniScroll`

moisesPC committed:
Prepare headers

moisesPC committed:
Set addresses

moisesPC committed:
Prepare honmaru_5f.c

moisesPC committed:
Finish .c

blazkowolf committed:
refactor(common): use macros and ternary in `mini_scroll_setup.c`

blazkowolf committed:
refactor(common): rename `{n,N}ecro_` functions

blazkowolf committed:
chore: depend on clangd

blazkowolf committed:
fix(common): use explicit cast to `Object*`

blazkowolf committed:
refactor(common): rename `{p,P}age_` functions

blazkowolf committed:
fix(common): rework some code and use explicit casts

blazkowolf committed:
refactor(object): rename `PauseMenuFuncI{d,D}`

blazkowolf committed:
refactor(overlay): rename `{p,P}auseMenu_` functions

blazkowolf committed:
fix(overlay): use `BITS` macros and explicit casts

blazkowolf committed:
fix: apply `void` to function prototypes where applicable

blazkowolf committed:
fix: clean up warnings with explicit casts

blazkowolf committed:
refactor(object): rename `textbox{A,_a}dvance{A,_a}rrow.h`

blazkowolf committed:
refactor(common): use `BITS` macros in `gameplay_menu_manager.c`

blazkowolf committed:
refactor(common): rename `{g,G}ameplayCommonTextbox_` functions

blazkowolf committed:
fix(common): use `BITS` macros and explicit casts

blazkowolf committed:
fix(common): use `BITS` macros and explicit casts

blazkowolf committed:
fix(effect): use `BITS` macros and explicit casts

blazkowolf committed:
refactor(overlay): rename `{o,O}peningBat_` functions

blazkowolf committed:
fix(overlay): use `BITS` macros and explicit casts

blazkowolf reviewed (approved):
This is an immense amount of great work by @moisesPC! I've made a ton of changes to use things like the `BITS` macros, adding explicit casts where implicit conversions would have been done, changing some of the control flow of code where it would have been deeply nested in `if`s (as far as matching would allow me), and extensive renaming of functions to follow our new conventions. Thank you for your great work @moisesPC!

moisesPC committed:
Match `src/overlay/map/honmaru_5f.c`

Merge pull request #48 from k64ret/honmaru_5f
